enhancements/metering/slim-install.md

```diff
@@ -0,0 +1,210 @@
+
+- `ReportDataSource`: exposes an existing table in Postgresql that can be used for reporting needs.
+- `ReportQuery`: defines a SQL query or view that references a `ReportDataSource` table, or a table created by a `Report` object.
+- `Report`: evaluates a templated SQL query defined in an existing `ReportQuery` custom resource and stores those query results in a Hive (now Postgres) table.
+
+At a high-level, the following must be done:
+
+- Create a `PostgresTable` CRD, which is an abstraction that represents an individual table in Postgres.
+- Remove (i.e. avoid reconciling) the internal `HiveTable` and `PrestoTable` custom resources.
+- Update the existing `ReportDataSource` promQL queries to reference the recording rules defined in the metering rule group.
+- Update the existing `ReportQuery` SQL queries and views to index Prometheus labels stored as `jsonb` objects in Postgres tables.
+
+### Implementation Details/Notes/Constraints [optional]
+
+#### Interacting with Postgres data types
+
+In order to conform with native Postgres data types, any Prometheus labels returned by the query_range API endpoint will need to be represented as the `jsonb` type. This could result in the following `PostgresTable` custom resource instantiation, where metric data is partitioned by datetime and the resultant table column name and types are explicitly defined:
+
+```yaml
+apiVersion: metering.openshift.io/v1
+kind: PostgresTable
+metadata:
+  name: reportdatasource-metering-node-allocatable-cpu-cores
+spec:
+  columns:
+  - name: amount
+    type: float8
+  - name: timestamp
+    type: timestamptz
+  - name: timeprecision
+    type: float8
+  - name: labels
+    type: jsonb
+  databaseName: metering
+  tableName: datasource_metering_node_allocatable_cpu_cores
+  managePartitions: false
+  partitionedBy:
+  - name: dt
+    type: string
+```
+
+Using a Postgresql go-client, the following implementation could be used to insert Prometheus metrics into an existing Postgresql table:
+
+```golang
+	_, err = conn.Exec(context.Background(), fmt.Sprintf("INSERT INTO %s VALUES($1, $2, $3, ($4)::jsonb)", tableName), metric.Amount, metric.Timestamp, metric.StepSize, labels)
+	if err != nil {
+		return err
+    }
+```
+
+#### Updating Default ReportDataSource and ReportQuery Objects
+
+The default, out-of-the-box `ReportDataSource` custom resources promQL queries will need to be updated in order to reference the recording rules that will be defined in the metering rule group. In order to do this, the metering-ansible-operator bundle will need to translate the existing set of promQL queries as a list of recording rules defined in the specification of a `PrometheusRule` custom resource. Support for [Operator bundles can include Prometheus-related custom resource objects](https://github.com/operator-framework/operator-lifecycle-manager/pull/1253) was added in the 0.14.1 OLM release.
```

Comment:
That's pretty awesome!

Reply:
what about user created(not out of the box) reportdatasources?

Reply:
> what about user created(not out of the box) reportdatasources?

still looking for an answer on this part.

Reply:
In the current state of the proposal, everything out-of-the-box uses recording rules and anything user-defined will need to be done on-demand when periodically pinging Prometheus for metrics. That could change once we get more clarification on whether dynamically creating recording rules for user-defined queries (i.e. recording rules not packaged in the metering-ansible-operator bundle) is a relatively safe operation for metering extension purposes.

Reply:
got it, thanks